job_executor/app.py

```diff
@@ -332,6 +333,8 @@ def _handle_manager_job(job: Job):
         datastore.remove(
             job_id, job.parameters.target, job.parameters.description
         )
+    elif operation == "ROLLBACK_REMOVE":
+        datastore.delete_draft(job_id, job.parameters.target)
     elif operation == "DELETE_DRAFT":
         datastore.delete_draft(job_id, job.parameters.target)
```

Comment:
I think it would be best to do some "typechecking here" by changing the delete_draft method to something like:
```py
def delete_draft(self, job_id: str, dataset_name: str, rollback_remove:bool):
        """
        Delete a dataset from the draft version of the datastore.
        """
        self._log(job_id, "initiated")
        job_service.update_job_status(job_id, "initiated")
        dataset_is_draft = self.draft_version.contains(dataset_name)
        dataset_operation = self.draft_version.get_dataset_operation(
            dataset_name
        )
        if dataset_operation != "REMOVE" and rollback_remove:
            log_message = f'"{dataset_name}" is not scheduled for removal'
            self._log(job_id, log_message, level="ERROR")
            job_service.update_job_status(job_id, "failed", log_message)
            return
        if (not dataset_is_draft) or (dataset_operation == "REMOVE" and not rollback_remove):
            log_message = f'Draft not found for dataset name: "{dataset_name}"'
            self._log(job_id, log_message, level="ERROR")
            job_service.update_job_status(job_id, "failed", log_message)
            return

        # If dataset has previously released data/metadata that needs to
        # be restored
        if dataset_operation in ["CHANGE", "PATCH_METADATA", "REMOVE"]:
            released_metadata = self.metadata_all_latest.get(dataset_name)
            if released_metadata is None:
                log_message = (
                    f"Can't find released metadata for {dataset_name} "
                    "when attempting to delete draft."
                )
                self._log(job_id, log_message, level="ERROR")
                raise VersioningException(log_message)
            self.metadata_all_draft.remove(dataset_name)
            self.metadata_all_draft.add(released_metadata)
        if dataset_operation == "ADD":
            self.metadata_all_draft.remove(dataset_name)
        if dataset_operation in ["ADD", "CHANGE"]:
            local_storage.delete_parquet_draft(dataset_name)
        self.draft_version.delete_draft(dataset_name)
        job_service.update_job_status(job_id, "completed")
```

and then calling these like:
```py
    elif operation == "ROLLBACK_REMOVE":
        datastore.delete_draft(job_id, job.parameters.target, rollback_remove=True)
    elif operation == "DELETE_DRAFT":
        datastore.delete_draft(job_id, job.parameters.target, rollback_remove=False)
```
I think this will eliminate a few tricky bugs that could occur in the future 🐛